Automattic/studio
# Studio: Add animated assistant icon

Fixes https://github.com/Automattic/dotcom-forge/issues/7525

## Proposed Changes

- Introduce an animated icon in the Assistant prompt input.
- Tweaked the styling of the input to accommodate the animated icon.

https://github.com/Automattic/studio/assets/417538/59396d04-9463-4152-bec1-6e05f96896a1

The animation is powered by the [Rive](https://rive.app) runtime. We pass some data directly to the runtime which then determines which states to show from the animation file.

This was a collaboration with Nebojsa Jurcic, who is responsible for the fantastic animations.

## Testing Instructions



- Run Studio using `STUDIO_AI=true npm start`
- Create a site (if you don't have one already)
- Go to the `Assistant` tab
- Note that the assistant icon pulses when the field is focussed.
- When typing the icon changes to a different state.
- If you stop typing, the icon reverts to the original state.
- When the assistant is thinking another state is shown.
- When complete, the icon reverts to the original state.

## Pre-merge Checklist
